instance/models/load_balancer.py

```diff
@@ -198,19 +198,24 @@ def run_playbook(self, ansible_vars):
         This is factored out into a separate method so it can be mocked out in the tests.
         """
         playbook_path = pathlib.Path(settings.SITE_ROOT) / "playbooks/load_balancer_conf/load_balancer_conf.yml"
-        with cache.lock("load_balancer_reconfigure:{}".format(self.domain), timeout=900):
-            returncode = ansible.capture_playbook_output(
-                requirements_path=str(playbook_path.parent / "requirements.txt"),
-                inventory_str=self.domain,
-                vars_str=ansible_vars,
-                playbook_path=str(playbook_path),
-                username=self.ssh_username,
-                logger_=self.logger,
-            )
+        returncode = ansible.capture_playbook_output(
+            requirements_path=str(playbook_path.parent / "requirements.txt"),
+            inventory_str=self.domain,
+            vars_str=ansible_vars,
+            playbook_path=str(playbook_path),
+            username=self.ssh_username,
+            logger_=self.logger,
+        )
         if returncode != 0:
             self.logger.error("Playbook to reconfigure load-balancing server %s failed.", self)
             raise ReconfigurationFailed
 
+    def _configuration_lock(self):
+        """
+        A Redis lock to protect reconfigurations of this load balancer instance.
+        """
+        return cache.lock("load_balancer_reconfigure:{}".format(self.domain), timeout=900)
```

Comment:
@haikuginger This timeout was already existing in the code.  Currently, reconfigurations take about two to three minutes, but with this change applied, we should be down to less than one minute, and we plan to further reduce the time, so I never expect this to become an issue.